src/deviceinfo.h

```diff
@@ -250,6 +251,14 @@ class SYSTEMSETTINGS_EXPORT DeviceInfo: public QObject
      */
     QStringList imeiNumbers();
 
+    /*!
+     * Unique device identifier
+     *
+     * Should be functionally equivalent with:
+     *   QDeviceInfo::uniqueDeviceID()
+     */
+    QString uniqueDeviceID();
```

Comment:
Unique ids can get tricky if something really starts considering them as definitive, unique and stable for all eternity. If this is now only used just as fallback on obex-capabilities if there are no imeis, thinking should we tone the promises down and e.g. call this "legacy" unique id, qtsystem uid, or something like that.

For another thing there's also SsuDeviceInfo::deviceUid() which is used in couple of places. Could obex-capabilities just use that or even figure out some id for itself (not sure does the returned value even matter anywhere)? Wouldn't sound good either if this and the ssu version returned different results.

Reply:
What ssu uses is: "imei | wlanmac" as just-digits string that is not  QUuid() compatible, whereas this is effectively "wlanmac|other-than-imei-things" as hyphenated-digits that is QUuid() compatible -> they return different result by design.

As to stability: there have been issues with both imeis and mac addresses. And in devices that have wlan but no cellular also ssu's id ends up being based on mac.

Then again, what obex-capability does with this is: it uses "imei|uniqueDeviceID" which is basically same as ssu's id thing as long we do not care about hyphenated vs not.

Also related: there is potentially related "serial" in about settings. Uses yet another list of possible sources with "TODO: eventually we should use QDeviceInfo's uniqueDeviceID()"

Now, if neither obex-capablity nor about settings needs this uniqueDeviceID(), we might as well drop this PR(*), use ssu's id for obex and remove that TODO from about settings.

*: or retain parts of it for getting mac address in about settings

Reply:
> Also related: there is potentially related "serial" in about settings. Uses yet another list of possible sources with "TODO: eventually we should use QDeviceInfo's uniqueDeviceID()"

Yea I'm not sure if that comment holds. Serial number can be a manufacturer specific thing that should resolve to very specific value while a uuid should be "just" unique and stable. Serial number should stay the same after reflashing, I'd assume, but up for debate whether that should be guaranteed for uuid.

> Now, if neither obex-capablity nor about settings needs this uniqueDeviceID(), we might as well drop this PR(*), use ssu's id for obex and remove that TODO from about settings.

Sounds good to me. Safer and easier for us if we don't need to go into the details of these ids and provide two a bit different forms.

> *: or retain parts of it for getting mac address in about settings

Yep, that info we will need.